django/db/models/expressions.py

```diff
@@ -149,6 +149,8 @@ class BaseExpression:
     _output_field_resolved_to_none = False
     # Can the expression be used in a WHERE clause?
     filterable = True
+    # Can the expression allowe null values?
+    nullable = True
```

Comment:
This flag should be set properly for all expressions, e.g. in `Col` should contain the same logic as in [`is_nullable()`](https://github.com/django/django/blob/2633c5341e94083dd802c5961b97b048adbd5940/django/db/models/sql/query.py#L2245-L2253), many text functions are not nullable on Oracle because it interprets empty strings as `NULL`'s, e.g. `MD5`, `Concat()` is not nullable, etc.